sencha-workspace/packages/slate-cbl/src/view/teacher/DashboardController.js

```diff
@@ -222,11 +242,145 @@ Ext.define('Slate.cbl.view.teacher.DashboardController', {
+            loadedCompetency = competenciesStore.getById(competencyCompletion.CompetencyID);
+            
+            if (!competencyProgressCell || !loadedCompetency) {
+                continue;
+            }
+
+            competencyPercent = Math.round(100 * (competencyCompletion.demonstrationsCount || 0) / loadedCompetency.get('totalDemonstrationsRequired'));
+
+            updatesCompetencies.push({
+                percent: competencyPercent,
+                average: competencyCompletion.demonstrationsAverage || 0,
+                isAverageLow: competencyCompletion.demonstrationsAverage < loadedCompetency.get('minimumAverage') && competencyPercent >= 50,
+                cellDom: competencyProgressCell,
+                textDom: Ext.fly(competencyProgressCell).down('.cbl-grid-progress-percent', true),
+                barDom: Ext.fly(competencyProgressCell).down('.cbl-grid-progress-bar', true),
+                averageDom: Ext.fly(competencyProgressCell).down('.cbl-grid-progress-average', true),
+                competency: loadedCompetency
+            });
+        }
+
+        // apply DOM writes for all updates in batch
+        Ext.Function.requestAnimationFrame(function() {
+            updatesLength = updatesCompetencies.length;
+            updateIndex = 0;
+
+            _compentencyDemonstrationResetFn = function(skill, skillIndex) {
+                for(var studentIndex in skill.demonstrationsByStudent) {
+                    skill.demonstrationsByStudent[studentIndex] = [];
+                }
+            };
+            
+            _updateCompetencySkillsFn = function(updatedDemonstrationSkills) {
+                // render details tables
+                updatedDemonstrationLength = updatedDemonstrationSkills.length;
+                updatedDemonstrationIndex = 0;
+                
+                for (; updatedDemonstrationIndex < updatedDemonstrationLength; updatedDemonstrationIndex++) {
+
+                    demonstratedSkill = updatedDemonstrationSkills[updatedDemonstrationIndex];
+                    skill = competencySkills.get(demonstratedSkill.SkillID);
+                    
+                    if (skill) {
+                        demonstrationsByStudent = skill.demonstrationsByStudent || (skill.demonstrationsByStudent = {});
+
+                        if (demonstratedSkill.StudentID in demonstrationsByStudent) {
+                            demonstrationsByStudent[demonstratedSkill.StudentID].push(demonstratedSkill);
+                        } else {
+                            demonstrationsByStudent[demonstratedSkill.StudentID] = [demonstratedSkill];
+                        }
+                    }
+
+                } 
+
+                skillStudentsTpl.overwrite(skillStudentsRow.down('tbody'), {
+                    skills: competencySkills.items,
+                    studentIds: Ext.Object.getKeys(update.competency.get('studentCompletions'))
```

Comment:
Yeah but it's got built-in falsey protection. Where Ext.[Type].foo polyfills are available I use them over native methods as they'll often smooth over edge issues in certain browsers, and those might be introduced in future browsers where they don't exist now